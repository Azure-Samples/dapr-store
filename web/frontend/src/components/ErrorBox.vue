<!--
// ----------------------------------------------------------------------------
// Copyright (c) Ben Coleman, 2020
// Licensed under the MIT License.
//
// Dapr Store frontend - Reusable error alert box component
// ----------------------------------------------------------------------------
-->

<template>
  <b-alert v-if="error" show variant="warning">
    <h4>There was a problem ðŸ˜¥</h4>
    <div class="errmsg">
      {{ error }}
    </div>
  </b-alert>
</template>

<script>
export default {
  name: 'ErrorBox',

  props: {
    error: {
      validator: (prop) => typeof prop === 'string' || typeof prop === 'object' || prop === null,
      required: true
    }
  },
}
</script>

<style scoped>
.errmsg {
  color: #222;
  font-family: 'Consolas', 'Courier New', Courier, monospace;
}
</style>