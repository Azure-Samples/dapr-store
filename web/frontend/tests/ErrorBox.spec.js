import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'

import ErrorBox from '@/components/ErrorBox.vue'

describe('ErrorBox.vue', () => {
  it('renders properly', () => {
    const wrapper = mount(ErrorBox, { props: { error: 'Really bad thing just happened' } })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
