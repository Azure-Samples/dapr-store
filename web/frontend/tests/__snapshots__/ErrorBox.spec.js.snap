// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorBox.vue > renders properly 1`] = `
"<div data-v-0e4d1e67="" class="alert alert-danger" show="">
  <h4 data-v-0e4d1e67="">There was a problem ğŸ˜¥</h4>
  <div data-v-0e4d1e67="" class="errmsg">Really bad thing just happened</div>
</div>"
`;
